### ISSUE TITLE
Pasting a date string loses the day

### ISSUE THREAD

**Issue by cadeyrn** [WordPress/gutenberg#28149]
**Describe the bug**

When using Firefox pasting a date in the format `XX. <month name> <Year>` loses the day, the result is `<month name> <Year>` without `XX.`

**To reproduce**

1. Make sure you have a date string in your clipboard, I tested with `18. Mai 2021` (German month name).
2. Start a new post in WordPress
3. Focus the first content block and press Strg + V (Mac: Cmd + V)

**Expected behavior**

You can read `18. Mai 2021`.

**Actual behavior**

You can read `Mai 2021`.

The browser console says:

```
Received HTML:

 <empty string> rich-text.min.js:2:44525
Received plain text:

 18. Mai 2021 rich-text.min.js:2:44568
Processed HTML piece:

 <ol start="18"><li>Mai 2021</li></ol>
```

When using the classic block there is no problem with pasting the date format.

**Editor version (please complete the following information):**
- WordPress version: 5.6
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? Answer: block editor that comes by default

**Desktop (please complete the following information):**
 - OS: macOS 11.1
 - Browser: Firefox 86 Nightly, Chrome 87 Stable

**Comment by annezazu** [comment]
@cadeyrn Thanks for the report! I tried to replicate this with WordPress 5.6 without any additional plugins installed and was unable to replicate. Do you have any other plugins installed that might be interfering?

**Comment by cadeyrn** [comment]
@annezazu Yes, I have several plugins installed so I tried again in another WordPress instance without plugins (I used this demo access: https://www.opensourcecms.com/wordpress/) and was able to reproduce the issue there as well.

**Comment by cadeyrn** [comment]
I made a screencast demonstrating the issue:

https://user-images.githubusercontent.com/1100614/104432790-a5794200-5589-11eb-9c9a-7aedb9460cb2.mov

**Comment by annezazu** [comment]
Ah ha! I can replicate this but only if I copy/paste from google search results :D How strange. If I copy and paste from this GitHub issue, it works just fine. Thank you for the video :)

**Comment by zvolanek** [comment]
I have same problem if I copy text from WordPad.

```
Received plain text:
18. May 2021 test
Processed HTML piece:
<ol start="18"><li>May 2021 test</li></ol>
```

```
Received plain text:
test 18. May 2021
Processed inline HTML:
test 18. May 2021
```

**Comment by R33D3M33R** [comment]
Hi,

here is a test sample pasted from plain text editor:

Pasted: 26. 11. 2021 - 18. 05. 2023
Result: 2021 - 18. 05. 2023

Result is pasted as **inline code**

If you omit the first space in the date it works: 26.11. 2021 - 18. 05. 2023

Please fix ASAP as it's extremely annoying.

**Comment by ellatrix** [comment]
This is the same as #33312, except here we do not want it to paste a list, but stay a paragraph.

**Comment by R33D3M33R** [comment]
Almost 5 years have passed since the first report and it still happens. Why can't the editor check if pasted item is a date, before trying to convert it to an ol?

Best regards

**Comment by Mamaduka** [comment]
Here's a proposed fix - #77949.
